### PR TITLE
[clang-tidy] rename designated initializers

### PR DESCRIPTION
**Pull request by revane**
readability-identifier-naming now supports renaming designated initializers.

### CODE REVIEW

File: clang-tools-extra/clang-tidy/utils/RenamerClangTidyCheck.cpp
Comment (EugeneZelenko) [inline clang-tools-extra/clang-tidy/utils/RenamerClangTidyCheck.cpp:374]: Please do not use `auto` unless type is explicitly stated in same statement or iterator. Should be also `const`.

File: clang-tools-extra/clang-tidy/utils/RenamerClangTidyCheck.cpp
Comment (EugeneZelenko) [inline clang-tools-extra/clang-tidy/utils/RenamerClangTidyCheck.cpp:377]: Ditto.

File: clang-tools-extra/clang-tidy/utils/RenamerClangTidyCheck.cpp
Comment (PiotrZSL) [inline clang-tools-extra/clang-tidy/utils/RenamerClangTidyCheck.cpp:382]: won't this getLocWithOffset point pass the end ?, usually end should point to last token, this will point to last token +1 character. In theory: `SourceRange FixLocation{Designator.getFieldLoc(), Designator.getFieldLoc());` should work in same way.

File: clang-tools-extra/test/clang-tidy/checkers/readability/identifier-naming.cpp
Comment (PiotrZSL) [inline clang-tools-extra/test/clang-tidy/checkers/readability/identifier-naming.cpp:765]: Add test (without whitespaces):
`Some_struct g_s{.SomeMember=1};`

**Review by PiotrZSL — COMMENTED** [review]
Release notes entry is missing. Should be:

"Improved [readability-identifier-naming](https://clang.llvm.org/extra/clang-tidy/checks/readability/identifier-naming.html) check in GetConfigPerFile mode by resolving symbolic links to header files. Fixed handling of Hungarian Prefix when configured to LowerCase. Added support for renaming designated initializers."

in clang-tools-extra/docs/ReleaseNotes.rst


Fix pointed out issues, and could land.

File: clang-tools-extra/clang-tidy/utils/RenamerClangTidyCheck.cpp
Comment (revane) [reply clang-tools-extra/clang-tidy/utils/RenamerClangTidyCheck.cpp]: You're right. Good to know!

File: clang-tools-extra/clang-tidy/utils/RenamerClangTidyCheck.cpp
Comment (EugeneZelenko) [inline clang-tools-extra/clang-tidy/utils/RenamerClangTidyCheck.cpp:374]: Should be `const`.

File: clang-tools-extra/clang-tidy/utils/RenamerClangTidyCheck.cpp
Comment (EugeneZelenko) [inline clang-tools-extra/clang-tidy/utils/RenamerClangTidyCheck.cpp:377]: Ditto.

File: clang-tools-extra/clang-tidy/utils/RenamerClangTidyCheck.cpp
Comment (revane) [reply clang-tools-extra/clang-tidy/utils/RenamerClangTidyCheck.cpp]: Sorry. I forgot that request.

File: clang-tools-extra/clang-tidy/utils/RenamerClangTidyCheck.cpp
Comment (HerrCai0907) [inline clang-tools-extra/clang-tidy/utils/RenamerClangTidyCheck.cpp:370]:
```suggestion
    for (const DesignatedInitExpr::Designator &Designator : Expr->designators()) {
```

**Review by HerrCai0907 — APPROVED** [review]
LGTM with a few minor change.
It would be clearer to rename the PR title / final merged comment with a more clearly directed description. e.g. 
`[clang-tidy] readability-identifier-naming supports renaming designated initializers`

**Comment by revane** [comment]
I updated the commit message but I can't remember if submitting a PR will use the PR description or the commit description in the submitted change. I'll update the PR description too just in case.

**Review by PiotrZSL — APPROVED** [review]
LGTM

**Comment by revane** [comment]
@AaronBallman Ready to go?

**Comment by AaronBallman** [comment]
> @AaronBallman Ready to go?

Yup, it was already accepted by the code owners. Based on our chat during office hours, I'm guessing you need help landing these changes. I'll go ahead and land them on your behalf. Feel free to let us know if we accept something and you can't land it yourself!

**Pull request merged**
